File: app/src/main/java/com/hawksjamesf/simpleweather/util/ActivityUtil.java
package com.hawksjamesf.simpleweather.util;

import android.app.Activity;
import android.content.Intent;

/**
 * Copyright ® $ 2017
 * All right reserved.
 *
 * @author: hawks.jamesf
 * @since: Oct/24/2018  Wed
 */
public class ActivityUtil {
    public static void openActivity(Activity activity, Class clazz, boolean finish) {
        activity.startActivity(new Intent(activity, clazz));
        if (finish) {
            activity.finish();
        }
    }

}

File: app/src/main/kotlin/com/hawksjamesf.simpleweather/util/ActivityUtil.kt
package com.hawksjamesf.simpleweather.util

import android.app.Activity
import android.content.Intent

/**
 * Copyright ® $ 2017
 * All right reserved.
 *
 * @author: hawks.jamesf
 * @since: Oct/24/2018  Wed
 */
inline fun <reified T : Activity> Activity.openActivity(finish: Boolean = true) {
    startActivity(Intent(this, T::class.java))
    if (finish) finish()
}

//typealias Callback:
//inline fun <reified T:Activity> Activity.openActivity(requestCode:Int,callback: Callback){
//    startActivityForResult(Intent(this,T::class.java),requestCode)
//}